cpp/src/parquet/size_statistics_test.cc

```diff
@@ -0,0 +1,307 @@
+  std::vector<int64_t> ref_levels;
+  std::vector<int64_t> def_levels;
+  std::vector<int64_t> byte_array_bytes;
+  bool operator==(const PageSizeStatistics& other) const {
+    return ref_levels == other.ref_levels && def_levels == other.def_levels &&
+           byte_array_bytes == other.byte_array_bytes;
+  }
+};
+
+class SizeStatisticsRoundTripTest : public ::testing::Test {
+ public:
+  void WriteFile(SizeStatisticsLevel level,
+                 const std::shared_ptr<::arrow::Table>& table) {
+    auto writer_properties = WriterProperties::Builder()
+                                 .max_row_group_length(2) /* every row group has 2 rows */
+                                 ->data_pagesize(1)       /* every page has 1 row */
+                                 ->enable_write_page_index()
+                                 ->enable_statistics()
+                                 ->set_size_statistics_level(level)
+                                 ->build();
+
+    // Get schema from table.
+    auto schema = table->schema();
+    std::shared_ptr<SchemaDescriptor> parquet_schema;
+    auto arrow_writer_properties = default_arrow_writer_properties();
+    ASSERT_OK_NO_THROW(arrow::ToParquetSchema(schema.get(), *writer_properties,
+                                              *arrow_writer_properties, &parquet_schema));
+    auto schema_node =
+        std::static_pointer_cast<schema::GroupNode>(parquet_schema->schema_root());
+
+    // Write table to buffer.
+    auto sink = CreateOutputStream();
+    auto pool = ::arrow::default_memory_pool();
+    auto writer = ParquetFileWriter::Open(sink, schema_node, writer_properties);
+    std::unique_ptr<arrow::FileWriter> arrow_writer;
+    ASSERT_OK(arrow::FileWriter::Make(pool, std::move(writer), schema,
+                                      arrow_writer_properties, &arrow_writer));
+    ASSERT_OK_NO_THROW(arrow_writer->WriteTable(*table));
+    ASSERT_OK_NO_THROW(arrow_writer->Close());
+    ASSERT_OK_AND_ASSIGN(buffer_, sink->Finish());
+  }
+
+  void ReadSizeStatistics() {
+    auto read_properties = default_arrow_reader_properties();
+    auto reader =
+        ParquetFileReader::Open(std::make_shared<::arrow::io::BufferReader>(buffer_));
+
+    // Read row group size statistics in order.
+    auto metadata = reader->metadata();
+    for (int i = 0; i < metadata->num_row_groups(); ++i) {
+      auto row_group_metadata = metadata->RowGroup(i);
+      for (int j = 0; j < metadata->num_columns(); ++j) {
+        auto column_metadata = row_group_metadata->ColumnChunk(j);
+        auto size_stats = column_metadata->size_statistics();
+        SizeStatistics row_group_stats;
+        if (size_stats != nullptr) {
+          row_group_stats = {size_stats->repetition_level_histogram,
+                             size_stats->definition_level_histogram,
+                             size_stats->unencoded_byte_array_data_bytes};
+        }
+        row_group_stats_.emplace_back(std::move(row_group_stats));
```

Comment:
Or more simply perhaps
```suggestion
        row_group_stats_.push_back(size_stats ? *size_stats : SizeStatistics{});
```